src/Storages/MergeTree/MergeList.cpp

```diff
@@ -40,6 +39,9 @@ MergeListElement::MergeListElement(
     {
         source_data_version = future_part->parts[0]->info.getDataVersion();
         is_mutation = (result_part_info.getDataVersion() != source_data_version);
+
+        WriteBufferFromString out(partition);
+        future_part->getPartition().serializeText(future_part->parts[0]->storage, out, {});
```

Comment:
Looks like this line is causing the exception. https://pastila.nl/?02aa875b/eb263ae20253222b7e4c9127675aa758

Reply:
```
Cannot parse expression of type Nullable(Int8) here: !Invalid                                                
: While executing ValuesBlockInputFormat
```

And the column right after `partition` is  `{"is_mutation", std::make_shared<DataTypeUInt8>()},` . This makes me suspect somewhere system.merges table schema is not aligned. causing the read results retrying to parse `partition` as `is_mutation`.

Reply:
Ah, we have another place defines the system.merges table. https://github.com/ClickHouse/ClickHouse/blob/55d228e78e977cf9c03d0a5dd922ade2f838b04a/tests/queries/0_stateless/02117_show_create_table_system.reference#L362

Don't understand what's it for, but I figured I should add partition column as well..